client/header/activity-panel/panels/stock/card.js

```diff
@@ -0,0 +1,152 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { BaseControl, Button } from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { withDispatch } from '@wordpress/data';
+
+/**
+ * WooCommerce dependencies
+ */
+import { Link, ProductImage } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import { ActivityCard } from '../../activity-card';
+
+class ProductStockCard extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			quantity: props.product.stock_quantity,
+			editing: false,
+		};
+
+		this.beginEdit = this.beginEdit.bind( this );
+		this.cancelEdit = this.cancelEdit.bind( this );
+		this.onQuantityChange = this.onQuantityChange.bind( this );
+		this.updateStock = this.updateStock.bind( this );
+	}
+
+	componentDidUpdate() {
+		this.quantityInput && this.quantityInput.focus();
+	}
+
+	beginEdit() {
+		this.setState( { editing: true } );
+	}
+
+	cancelEdit() {
+		this.setState( {
+			editing: false,
+			quantity: this.props.product.stock_quantity,
+		} );
+	}
+
+	onQuantityChange( quantity ) {
+		this.setState( { quantity } );
+	}
+
+	updateStock() {
+		const { product, updateItem } = this.props;
+
+		this.setState( { editing: false }, () => {
+			const data = {
+				stock_quantity: this.state.quantity,
+				type: product.type,
+				parent_id: product.parent_id,
+			};
+
+			updateItem( 'products', product.id, data );
+		} );
+	}
+
+	getActions() {
+		const { editing } = this.state;
```

Comment:
Design nitpick here: I always find it satisfying when a change in mode results in the same height as the previous, so we can prevent the elements below from jumping. 

![jump](https://user-images.githubusercontent.com/1922453/55600208-8bf17680-57b7-11e9-9432-9aa03eceff1e.gif)

Reply:
@josemarques @LevinMedia what's your take on this design issue?

Reply:
Let's capture this for a follow-up issue / question.

Reply:
@jeffstieler I agree.